### PR TITLE
Fix the required fields of event form

### PR DESCRIPTION
**Pull request by priyesh777**
## Addresses :- 

- https://github.com/idmc-labs/Helix2.0/issues/207#issue-952243761
- https://github.com/idmc-labs/Helix2.0/issues/205
- https://github.com/idmc-labs/Helix2.0/issues/134

## Depends:-
- https://github.com/idmc-labs/helix-server/pull/211

## Changes
- Add filters and columns as per required in issue #207 
- Add sorting to columns of Crises/Event/Country
- Add DisasterCategory column and ConflictCategory column in Events table

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: src/components/tables/EntriesTable/EntriesFilter/index.tsx [hunk 1]
```diff
@@ -123,18 +98,6 @@ function EntriesFilter(props: EntriesFilterProps) {
                     onChange={onValueChange}
                     placeholder="Search"
                 />
-                <MultiSelectInput
```
Comment (tnagorra) [inline src/components/tables/EntriesTable/EntriesFilter/index.tsx:126]: Why remove this?

File: src/views/Events/EventsFilter/index.tsx [hunk 2]
```diff
@@ -53,6 +88,8 @@ const schema: FormSchema = {
         crisisByIds: [arrayCondition],
         name: [],
         glideNumber_Icontains: [],
+        violenceTypes: [nullCondition],
+        disasterCategories: [nullCondition],
```
Comment (tnagorra) [inline src/views/Events/EventsFilter/index.tsx:92]: Why pass nullCondition for violenceTypes and disasterCategories?

File: src/views/Report/index.tsx [hunk 3]
```diff
@@ -566,7 +569,8 @@ function Report(props: ReportProps) {
     const reportPermissions = user?.permissions?.report;
 
     const report = reportData?.report;
-
+    const conflictType = report?.eventsReport?.results?.find((item) => item.eventType === 'CONFLICT');
+    const disasterType = report?.eventsReport?.results?.find((item) => item.eventType === 'DISASTER');
```
Comment (tnagorra) [inline src/views/Report/index.tsx:573]: We should get this from filter not the actual report data.

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/Report/index.tsx (hunk 3 above)
Comment (priyesh777) [reply src/views/Report/index.tsx:573]: This was intended to show dis-aggregated data of a particular report based upon whether it is a "CONFLICT" or "DISASTER", this change is for the issue: https://github.com/idmc-labs/Helix2.0/issues/205

Comment (tnagorra) [reply src/views/Report/index.tsx:574]: I know. But the way to identify conflict and disaster type report is not correct

File: src/views/Events/EventsFilter/index.tsx [hunk 4]
```diff
@@ -126,8 +179,29 @@ function EventsFilter(props: EventsFilterProps) {
...
+                    disasterCategoryId: disasterCategory.id,
+                    disasterCategoryName: disasterCategory.name,
+                }))
+            ))
+        ))
+    )).filter(isDefined);
+
+    const conflictType = value.eventTypes?.includes('CONFLICT');
+    const disasterType = value.eventTypes?.includes('DISASTER');
+
```
Comment (tnagorra) [inline src/views/Events/EventsFilter/index.tsx:204]: Also add this check in schema

File: src/views/Report/index.tsx [hunk 5]
```diff
@@ -566,7 +569,7 @@ function Report(props: ReportProps) {
     const reportPermissions = user?.permissions?.report;
 
     const report = reportData?.report;
-
+    const reportType = report?.filterEventCrisisTypes?.toLocaleString();
```
Comment (tnagorra) [inline src/views/Report/index.tsx:572]: Why use toLocaleString?

File: src/views/Report/index.tsx [hunk 6]
```diff
@@ -709,46 +712,54 @@ function Report(props: ReportProps) {
...
-                                label={(
-                                    <>
-                                        No. of IDPs
-                                        <br />
-                                        (Disaster)
-                                    </>
-                                )}
-                                value={report?.totalDisaggregation?.totalStockDisasterSum}
-                            />
+                            {reportType && reportType === 'CONFLICT' && (
```
Comment (tnagorra) [inline src/views/Report/index.tsx:715]: Add this check to form schema

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/Report/index.tsx (hunk 5 above)
Comment (priyesh777) [reply src/views/Report/index.tsx:572]: image
The report type occurs in this format, due which converting to LocaleString was identified as a simpler solution.

File: src/views/Report/index.tsx (hunk 6 above)
Comment (priyesh777) [reply src/views/Report/index.tsx:715]: There is no schema defined for the particular use case.

File: src/views/Report/index.tsx (hunk 5 above)
Comment (tnagorra) [reply src/views/Report/index.tsx:572]: Better to use `Array.join(', ')`

File: src/views/Report/index.tsx (hunk 6 above)
Comment (tnagorra) [reply src/views/Report/index.tsx:715]: There should be a schema defined

**Pull request merged**